Replace strip_heredoc with squiggly

ibrahima opened the pull request:
* Just replaces usages of Rails' strip_heredoc with Ruby 2.3's squiggly heredoc: https://infinum.co/the-capsized-eight/multiline-strings-ruby-2-3-0-the-squiggly-heredoc

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:

### Risks
- Level: Low

grosser commented on app/models/changeset/pull_request.rb:
```diff
@@ -9,6 +9,9 @@ class Changeset::PullRequest
   RISKS_SECTION = /^\s*#*\s*Risks?\s*#*\s*\n(?:\s*[-=]*\s*\n)?/i
 
+  # Matches a markdown section heading
+  SECTION_HEADING = /^\s*#*\s*\w+.*\n(?:\s*[-=]*\s*\n)?/i
```
this does not belong here

grosser commented:
rebase ... otherwise looks good

ibrahima commented:
Oh, I just based it on my other PR since that one adds heredocs, but I can just change the ones in that PR there and then have this one for the existing strings. Rebased and removed those commits!

ibrahima commented:
Not sure if that was meant for me, but I don't have write access, I'm not from Zendesk - I've just submitted a few PRs before. But thanks for accepting my changes!

grosser commented:
oh, it said `Contributor` ... but we are hiring anyways ... so you can always change that ;)

ibrahima commented:
Looks like Brakeman doesn't understand this change somehow? https://github.com/zendesk/samson/pull/2031/files#diff-b17d53435fe1389ba1c27dd996b4c93e

Does it need upgrading to support Ruby 2.3?

grosser commented:
needs https://github.com/seattlerb/ruby_parser/issues/218 ... so I guess this will sit here for a while :(

... could do some gsub hacks in brakeman ... but feels dirty ... don't need it that bad ...

ibrahima commented:
I could also just revert that one change, and you'd have the new syntax elsewhere? Or will Brakeman break on other files too? I guess it's not specific to that file, but some combination of syntaxes that causes the parser error.

It definitely handles some form of squiggly heredocs because our codebase has them and we also have Brakeman enabled.

grosser commented:
I'd thin kit ignores them because the syntax is still somehow valid ... but
most likely broken ... I'll revert the ones that got merged manually so we
don't skip any brakeman warnings by accident

On Tue, Jun 6, 2017 at 3:05 PM, Ibrahim Awwal <notifications@github.com>
wrote:

> I could also just revert that one change, and you'd have the new syntax
> elsewhere? Or will Brakeman break on other files too? I guess it's not
> specific to that file, but some combination of syntaxes that causes the
> parser error.
>
> It definitely handles some form of squiggly heredocs because our codebase
> has them and we also have Brakeman enabled.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/2031#issuecomment-306631122>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ8lzqz62Nl5EnFbHotMNU20EnwSWks5sBc0qgaJpZM4Nx4FN>
> .
>

grosser commented:
ah these were in tests ... so that's ok since brakeman does not check them
